src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -58,12 +58,38 @@ public byte[] mask(byte[] input) {
      */
     private void visitValue(MaskingState maskingState, @CheckForNull KeyMaskingConfig keyMaskingConfig) {
         skipWhitespaceCharacters(maskingState);
-        if (AsciiCharacter.isSquareBracketOpen(maskingState.byteAtCurrentIndex())) {
-            visitArray(maskingState, keyMaskingConfig);
-        } else if (AsciiCharacter.isCurlyBracketOpen(maskingState.byteAtCurrentIndex())) {
-            visitObject(maskingState, keyMaskingConfig);
-        } else {
-            visitPrimitive(maskingState, keyMaskingConfig);
+        switch (maskingState.byteAtCurrentIndex()) {
+            case '[' -> visitArray(maskingState, keyMaskingConfig);
+            case '{' -> visitObject(maskingState, keyMaskingConfig);
+            case '-', '+', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.' -> {
```

Comment:
I don't think a number in valid JSON can start with a `.`  or a `+`.

Only `0-9` and `-` 

If your intent was to match on all numeric characters, you are missing `E` and `e`

Reply:
Done